src/Assignments/ProblemIframe.tsx

```diff
@@ -60,42 +89,46 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         setRenderedHTML('');
         // srcdoc='' triggers onLoad with setLoading(false) so setLoading(true) isn't effective until now
         setLoading(true); 
-        (async () => {
-            try {
-                let queryString = qs.stringify(_({
-                    workbookId,
-                    readonly
-                }).omitBy(_.isUndefined).value());
-                if (!_.isEmpty(queryString)) {
-                    queryString = `?${queryString}`;
-                }
 
-                let res;
-                if (previewPath || previewProblemSource) {
-                    res = await postPreviewQuestion({
-                        webworkQuestionPath: previewPath,
-                        problemSeed: previewSeed,
-                        problemSource: previewProblemSource,
-                        showHints: previewShowHints,
-                        showSolutions: previewShowSolutions
-                    });
-                } else {
-                    res = await AxiosRequest.get(`/courses/question/${problem.id}${queryString}`);
-                }
-                // TODO: Error handling.
-                setRenderedHTML(res.data.data.rendererData.renderedHTML);
-            } catch (e) {
-                setError(e.message);
-                logger.error('Error posting preview', e, e.message);
-                setLoading(false);
-            }
-        })();
+        fetchHTML();
```

Comment:
Why did we move this to another function if it is never reused and also scoped so that it can't be reused?

Reply:
Just on that point, it's a good thing to break code out into separate functions, even if they're not reusable. If helps readability and improves cyclomatic complexity scores.

Reply:
That's a great point, I like it

Still feels like we should move it out of the really finite scope of useEffect, no reason it has to be in here